### PR TITLE
enhancement: upgrade to laravel 11

### LINKED ISSUE #10: [BE] Upgrade the project to Laravel 11
### Task Requirements
We can upgrade the API to Laravel 11 when it comes out on March 12 2024.

[Follow the upgrade guide](https://laravel.com/docs/11.x/upgrade) to do the upgrade.

### PR DESCRIPTION
**Pull request by jovangjokovski**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -5,10 +5,10 @@
     "keywords": ["laravel", "framework"],
     "license": "MIT",
     "require": {
-        "php": "^8.3",
+        "php": "^8.2.4",
```
Comment (davorminchorov) [inline composer.json]: This change can be reverted. We are enforcing environments to use PHP 8.3 everywhere.

**Comment by davorminchorov** [comment]
Also, don't forget about [Updating Sanctum](https://laravel.com/docs/11.x/upgrade#updating-sanctum).

**Comment by y0vche** [comment]
> Also, don't forget about [Updating Sanctum](https://laravel.com/docs/11.x/upgrade#updating-sanctum).

Sanctum part is updated automatically by composer. The piece of code in `config/sanctum.php` file, is updated as per documentation.

File: composer.json (hunk 1 above)
Comment (y0vche) [reply composer.json]: Latest PHP version on XAMPP is 8.2.4. I will move to Herd for the upcoming tasks. Pushed the change. There should be no conflicts as we still haven't used any of php 8.3 features.

**Review by davorminchorov — APPROVED** [review]

**Pull request merged**